akka-stream/src/main/scala/akka/stream/scaladsl/JsonFraming.scala

```diff
@@ -57,8 +58,8 @@ object JsonFraming {
           tryPopBuffer()
 
         override def onUpstreamFinish(): Unit = {
-          try buffer.poll() match {
-            case Some(json) ⇒ emit(out, json, () ⇒ completeStage())
+          buffer.poll() match {
+            case Some(json) ⇒ emit(out, json)
```

Comment:
#21150 stage completed after emitting one element when upstream closes (even if there was more elements available in the buffer)